File: 80.removeDuplicates.go
package main

func removeDuplicates(nums []int) int {
	size := len(nums)

	if size <= 2 {
		return size
	}

	wptr := 2
	rptr := 2

	for rptr < size {
		if nums[rptr] != nums[wptr-2] {
			nums[wptr] = nums[rptr]
			wptr++
		}
		rptr++
	}

	return wptr
}

File: 80.removeDuplicates.c
#include "shdstd.h"

int removeDuplicates(int *nums, int size)
{
	if (size <= 2) {
		return size;
	}

	int wptr = 2;
	int rptr = 2;

	while (rptr < size) {
		if (nums[rptr] != nums[wptr-2]) {
			nums[wptr++] = nums[rptr];
		}
		++rptr;
	}

	return wptr;
}
